packages/openstef-models/src/openstef_models/feature_engineering/weather_transforms/air_related_features_transform.py

```diff
@@ -0,0 +1,196 @@
+# SPDX-FileCopyrightText: 2025 Contributors to the OpenSTEF project <short.term.energy.forecasts@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""Transform for calculating and adding air-related meteorological features to a time series dataset.
+
+The transform computes saturation vapour pressure, vapour pressure, dewpoint, and air density
+based on temperature, pressure, and relative humidity columns using established physical equations.
+"""
+
+from typing import Any, Literal, override
+
+import numpy as np
+import pandas as pd
+from pydantic import Field
+
+from openstef_core.base_model import BaseConfig
+from openstef_core.datasets import TimeSeriesDataset
+from openstef_core.datasets.transforms import TimeSeriesTransform
+from openstef_core.exceptions import MissingColumnsError
+
+type AirRelatedFeatureName = Literal["saturation_vapour_pressure", "vapour_pressure", "dewpoint", "air_density"]
+
+
+class AirRelatedFeaturesTransform(BaseConfig, TimeSeriesTransform):
+    """Transform that calculates air-related meteorological features from basic weather data.
+
+    This transform calculates various air-related features including saturation vapour pressure,
+    vapour pressure, dewpoint, and air density using standard meteorological formulas. It requires
+    temperature, pressure, and relative humidity as input columns.
+    The calculated features can be used to enhance weather-based prediction models by providing
+    additional atmospheric state information that may correlate with energy consumption patterns.
```

Comment:
This part is quite vague. I think we can write the actual reason. As far as I know air pressure and vapour affects PV generation. There were a few papers about this, citation needed. But here is what google says:


> Higher air pressure, often found at lower altitudes, tends to increase solar panel output (voltage, current, and intensity) because it creates a stronger downward pull on solar radiation. Conversely, lower air pressure at high altitudes has the opposite effect, potentially decreasing performance and voltage requirements.